### PR TITLE
Ensuring that Vue ajax-select Components are mounted when added in multiple select fields for Numismatic Issues

### LINKED ISSUE #3666: Numismatics reference ajax dropdown doesn't work with multiple nested resources
Screen Shot 2020-02-12 at 11 35 04 AM

When clicking "add another citation" on the issue form, the "Numismatics reference" dropdown doesn't appear.

Also, it was requested to move the reference autocomplete field above the part and number fields

### PR DESCRIPTION
**Pull request by jrgriffiniii**
I am still testing this, but given that this seems to be a higher priority issue, I thought that I should offer this in order to expedite requesting input.

Resolves #3666

### CODE REVIEW

**Comment by jrgriffiniii** [comment]
This is working for me locally, I've just been struggling with implementing feature tests (the Capybara/Selenium element selectors are not functioning as expected).

File: spec/resources/numismatics/issues/numismatics/issues_feature_spec.rb [hunk 1]
```diff
@@ -196,6 +196,58 @@
...
+
+    before do
+      reference1
+      reference2
+    end
+
+    scenario "when users add citations for an Issue", js: true do
+      visit edit_numismatics_issue_path(numismatic_issue)
+
+      # find("#numismatic_citation input.vs__search").send_keys("Test Reference", :enter)
```
Comment (jrgriffiniii) [inline spec/resources/numismatics/issues/numismatics/issues_feature_spec.rb]: I could not manage to find any way to trigger this behavior - I spent some time attempting various combinations of CSS selectors, using Nokogiri to isolate the DOM elements, and using jQuery through https://rubydoc.info/github/jnicklas/capybara/master/Capybara/Session#evaluate_script-instance_method

I can remove these statements (or tests entirely), as I am not quite certain how to proceed with testing for this.

File: spec/resources/numismatics/issues/numismatics/issues_feature_spec.rb [hunk 2]
```diff
@@ -196,6 +196,58 @@
...
+
+    before do
+      person1
+      person2
+    end
+
+    scenario "when users add artists for an Issue" do
+      visit edit_numismatics_issue_path(numismatic_issue)
+
+      # find("#numismatic_artist input.vs__search").send_keys("name1", :enter)
```
Comment (jrgriffiniii) [inline spec/resources/numismatics/issues/numismatics/issues_feature_spec.rb]: Please see the comment above.

**Comment by jrgriffiniii** [comment]
This isn't completely ready to be merged, however, I would need to please request a review in response to https://github.com/pulibrary/figgy/pull/3680/files#r379537180

File: spec/resources/numismatics/issues/numismatics/issues_feature_spec.rb (hunk 1 above)
Comment (eliotjordan) [reply spec/resources/numismatics/issues/numismatics/issues_feature_spec.rb]: Maybe remove the statements. Do you think the tests are useful without them? If not, I might remove them as well.  I have found doing JS testing of these components with Capybara nearly impossible.

Comment (jrgriffiniii) [reply spec/resources/numismatics/issues/numismatics/issues_feature_spec.rb]: I can remove the statements, thank you for reviewing this.

There is one approach I might be able to propose for future PRs, using https://jestjs.io/docs/en/puppeteer, but I will delay this for future work.

**Pull request merged**